sequencer/src/header.rs

```diff
@@ -183,6 +192,23 @@ impl Header {
             transactions_root,
             fee_merkle_tree_root,
             block_merkle_tree_root,
+            fee_info: parent.fee_info,
+            builder_signature: None,
+        };
+
+        // Sign Header with builder wallet from state
+        let signing_key: &SigningKey = builder_address.signer();
+        let header_signature: ecdsa::Signature = signing_key.sign(header.commit().as_ref());
+        // In order to use `type::Signature` for verification we do a
+        // `TryFrom` conversion on the signature. But first we need to
+        // append Etherium's `Electurm` value (can be 27 or 28). There
+        // must be a better way to do this.
```

Comment:
There does not seem to be a better way to do this. I can't believe ethers doesn't just have a function to sign arbitrary bytes and get a `Signature` 🙄 

Anyways, I think the choice of `v` is not arbitrary (has to be either 27 or 28 depending on the signing key). We can a different overloading of `sign` which gives an `ecdsa::Signature` and a `RecoveryId`  (which tells you what `v` is supposed to be) and then copy what ethers does to reconstruct the `types::Signature` [here](https://docs.rs/ethers-signers/2.0.13/src/ethers_signers/wallet/mod.rs.html#152)

Reply:
This might be why there's no such functionality in ethers-rs: https://github.com/gakonst/ethers-rs/pull/2369#issuecomment-1520607943

Reply:
Ah, I guess that decision makes sense. In our case it should be fine, since we're not signing arbitrary data, but it's up to the software to be sure of that. We may have to revisit this decision in the future if we ever want to support hardware signers, but that's probably a ways off